packages/reakit/src/Popover/__examples__/HoverPopover/index.tsx

```diff
@@ -0,0 +1,107 @@
+import * as React from "react";
+import { Button } from "reakit/Button";
+import {
+  usePopoverState,
+  Popover,
+  PopoverDisclosure,
+  PopoverArrow,
+  PopoverInitialState,
+} from "reakit/Popover";
+import { Separator } from "reakit/Separator";
+import { VisuallyHidden } from "reakit/VisuallyHidden";
+
+function useHoverPopoverState(initialState?: PopoverInitialState) {
+  const timeout = 300;
+  const showTimeout = React.useRef<number | null>(null);
+  const hideTimeout = React.useRef<number | null>(null);
+  const popover = usePopoverState(initialState);
+  const clearTimeouts = React.useCallback(() => {
+    if (showTimeout.current !== null) {
+      window.clearTimeout(showTimeout.current);
+    }
+    if (hideTimeout.current !== null) {
+      window.clearTimeout(hideTimeout.current);
+    }
+  }, []);
+  const show = React.useCallback(() => {
+    clearTimeouts();
+    showTimeout.current = window.setTimeout(() => {
+      popover.show();
+    }, timeout);
+  }, [clearTimeouts, popover.show]);
+  const hide = React.useCallback(() => {
+    clearTimeouts();
+    hideTimeout.current = window.setTimeout(() => {
+      popover.hide();
+    }, timeout);
+  }, [clearTimeouts, popover.hide]);
+  React.useEffect(
+    () => () => {
+      clearTimeouts();
+    },
+    [clearTimeouts]
+  );
+  return {
+    ...popover,
+    show,
+    hide,
+  };
+}
+
+type User = {
+  fullname: string;
+  username: string;
+  description: string;
+};
+
+const user: User = {
+  fullname: "John Doe",
+  username: "@JohnDoe",
+  description: "Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
+};
+
+export default function HoverPopover() {
+  const [isFollowing, setFollowing] = React.useState(false);
+  const popover = useHoverPopoverState();
+  const popoverProps = {
+    onMouseEnter: popover.show,
+    onMouseLeave: popover.hide,
+  };
+  return (
+    <>
+      <PopoverDisclosure
+        as="a"
+        href="#"
+        aria-label={`See the complete profile of ${user.username}`}
+        {...popover}
+        {...popoverProps}
+      >
+        {user.username}
+      </PopoverDisclosure>
+      <VisuallyHidden>
+        <PopoverDisclosure {...popover}>
+          See the preview of the profile of {user.username}
+        </PopoverDisclosure>
+      </VisuallyHidden>
```

Comment:
>you can't show the popover using the keyboard without actually activating the link, which would flash the popover and then navigate to the link address.

Not really, it's a second button, but yes is visually hidden!
But I agree, having to perform a click is annoying...

>I imagine this as an arrow down or ellipsis button that appears floating next to the link when it gets focus. In the future, we may introduce some kind of useFocusVisible helper so we only show this button when the link gets keyboard focus, and not when it gets clicked (created #737).

That's really interesting! 
I'm wondering, what do you think about introducing this kind of feature on VisuallyHidden directly?
I mean it could be useful for Skip Links, for example, 
Imagine, a VisuallyHidden which simply becomes Visible when it has the focus? 🤷‍♂️

Reply:
They're different things. A `SkipToContent` kind of component would receive focus when using the keyboard, whereas `VisuallyHidden` would only receive focus when using a screen reader. `VisuallyHidden` is inaccessible to sighted keyboard users.

And showing the button when it gets focus is not really what we want here. We want to show it when another element gets focus (in this case, the username link), so both keyboard and screen reader users could focus on it by pressing <kbd>Tab</kbd>.

The only problem with this approach I'm suggesting is that the hidden disclosure would also appear when a mouse user clicks on the link cause it would get focus. This is not a big deal as this is just an example. If we come up with a `HoverCard` component in the future though, we should address this. And we could use the `onFocusVisible` prop that I've suggested on #737.

Reply:
Oh god, I didn't get it at all, thank you for the clarification 👌

Reply:
But this is more complicated than that: if we show/hide the disclosure button when the link gets focused/blurred, the disclosure would disappear when we move focus to it (because the link got blurred). We should then check if the disclosure has focus. But when the focus is moved into the popover, we should also keep showing the disclosure.

Reply:
The fact that it's been challenging to come up with this example is a clear signal that we need a `HoverCard` component. 😅

Reply:
Definitively, indeed.

So, what's next? I try to fix it now or I close it? WDYT?

Reply:
This PR is really valuable as you're experimenting with things that may become the `HoverCard` component in the future. I would like to keep it open for this reason. If you enjoy working on this, please keep going. Otherwise, no pressure. ❤️

Reply:
So the targeted JSX is something like that, isn't it?
```jsx
const hoverCard = useHoverCardState();
<HoverCardDisclosure {...hoverCard} as="a" href="/users/diegohaz">@diegohaz</HoverCardDisclosure>
<HoverCard {...hoverCard}>Profile overview here</HoverCard>
```

Reply:
Yeah, kinda! This is what I have in mind based on what we've discussed in this PR:
```jsx
const hoverCard = useHoverCardState();

// Component that will receive only onMouseEnter and onMouseLeave props. No aria-attributes.
<HoverCardReference {...hoverCard} as="a" href="/users/diegohaz">@diegohaz</HoverCardReference>

// Component that will be visible when HoverCardReference gets focused and hidden when all the HoverCard* components, including HoverCard children, get blurred. It's going to have the same semantics as PopoverDisclosure
<HoverCardDisclosure {...hoverCard} />

<HoverCard {...hoverCard}>Profile overview here</HoverCard>
```

Reply:
Crystal clear, I will try as an example for now then! 💯